cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -0,0 +1,354 @@
+        message='Waiting for hourly rollups')
+
+    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
+    # in the metric_rollups DB table.The values from these hourly records are summed up.
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        consumed_hours = consumed_hours + 1
+        if (record.cpu_usagemhz_rate_average or
+           record.cpu_usage_rate_average or
+           record.derived_memory_used or
+           record.net_usage_rate_average or
+           record.disk_usage_rate_average):
+            average_cpu_used_in_mhz = average_cpu_used_in_mhz + record.cpu_usagemhz_rate_average
+            average_memory_used_in_mb = average_memory_used_in_mb + record.derived_memory_used
+            average_network_io = average_network_io + record.net_usage_rate_average
+            average_disk_io = average_disk_io + record.disk_usage_rate_average
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        if record.derived_vm_used_disk_storage:
+            average_storage_used = average_storage_used + record.derived_vm_used_disk_storage
+
+    # Convert storage used in Bytes to GB
+    average_storage_used = average_storage_used * math.pow(2, -30)
+
+    return {"cpu_used": average_cpu_used_in_mhz,
+            "memory_used": average_memory_used_in_mb,
+            "network_io": average_network_io,
+            "disk_io_used": average_disk_io,
+            "storage_used": average_storage_used,
+            "consumed_hours": consumed_hours}
+
+
+@pytest.yield_fixture(scope="module")
+def metering_report(vm_ownership, provider):
+    # Create a Metering report based; Queue the report.
+    owner = vm_ownership
+    data = {
+        'menu_name': 'cb_' + provider.name,
+        'title': 'cb_' + provider.name,
+        'base_report_on': 'Metering for VMs',
+        'report_fields': ['Owner', 'Memory Used',
+        'CPU Used', 'Disk I/O Used',
+        'Network I/O Used', 'Storage Used',
+        'Existence Hours Metric', 'Metering Used Metric'],
+        'filter': {
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'
+        }
+    }
+    report = CustomReport(is_candu=True, **data)
+    report.create()
+
+    logger.info('Queuing chargeback report with default rate for {} provider'.format(provider.name))
+    report.queue(wait_for_finish=True)
```

Comment:
**Required**

In 5.9 this navigation is failing, the report name isn't found in the BootstrapTreeview

Reply:
Yes, this is because of this BZ https://bugzilla.redhat.com/show_bug.cgi?id=1531600 
Custom reports not visible on the UI